intro-HPC/ch_checkpointing.tex

```diff
@@ -1,4 +1,171 @@
+or additional output files created by the job in its working directory are required,
+the \lstinline|--post| flag should be used. This will copy the entire job working directory
+to the location where \lstinline|csub| was executed, in a directory named \lstinline|result.<jobname>|.
+An alternative is to copy the interesting files to the shared storage at the end of the job script.
+
+
+\subsection{Running on shared storage (\texttt{--shared})}
+
+If the job needs to be run on the shared storage, \lstinline|--shared| should be specified.
+You should enable this option by default, because it makes the execution of the
+underlying \lstinline|csub| script more robust: it doesn't have to copy from/to
+the local storage on the workernode. When enabled, the job will be run in a
+subdirectory of \lstinline|$VSC_SCRATCH/chkpt|. All files produced by the job will
+be in \lstinline|$VSC_SCRATCH/chkpt/<jobid>/| while the job is running.
+
+Note that if files in the directory where the job script is located are required to
+run the job, you should also use \lstinline|--pre|.
+
+\subsection{Job wall time (\texttt{--job\_time}, \texttt{--chkpt\_time})}
+
+To specify the requested wall time per subjob, use the \lstinline|--job-time| parameter.
+The default setting is 10 hours per (sub)job. Lowering this will result in more
+frequent checkpointing, and thus more (sub)jobs.
+
+To specify the time that is reserved for checkpointing the job, use \lstinline|--chkpt_time|.
+By default, this is set to 15 minutes which should be enough for most applications/jobs.
+\emph{Don't change this unless you really need to}.
+
+The total requested wall time per subjob is the sum of both \lstinline|job_time| and \lstinline|chkpt_time|.
+
+If you would like to time how long the job executes, just prepend the main command in your job script with
+the time command \lstinline|time|, e.g.:
+
+\begin{prompt}
+time main_command
+\end{prompt}
+
+The \lstinline|real| time will not make sense, as it will also include the time
+passed between two checkpointed subjobs. However, the \lstinline|user| time should give a good
+indication of the actual time it took to run your command, even if multiple checkpoints were performed.
+
+\subsection{Resuming from last checkpoint (\texttt{--resume})}
+
+The \lstinline|--resume| option allows you to resume a job from the
+last available checkpoint in case something went wrong
+(e.g. accidentally deleting a (sub)job using \lstinline|qdel|, a power outage or other system failure, \ldots).
+
+\the\year\the\month\the\day
+
+Specify the job name as returned after submission (and as listed in \texttt{\$VSC\_SCRATCH/chkpt}).
+The full job name consists of the specified job name (or the script name if no job name was specified),
+a timestamp and two random characters at the end, for example \texttt{my\_job\_name.\the\year0102\_133755.Hk}
+or \texttt{script.sh.\the\year0102\_133755.Hk}.
```

Comment:
let's not hardcode `0102` either?

Reply:
It's not easy to zero-pad numbers in latex (because `\the\month` can output a single digit).

And this is the pragmatic solution: it's maximum a year behind and only "wrong" 2 days in a year ;)

Reply:
OK, good enough